viewer/packages/camera-manager/src/Flexible/FlexibleCameraManager.ts

```diff
@@ -0,0 +1,551 @@
+import { FlexibleControlsType } from './FlexibleControlsType';
+import { FlexibleMouseActionType } from './FlexibleMouseActionType';
+import { DebouncedCameraStopEventTrigger } from '../utils/DebouncedCameraStopEventTrigger';
+import { FlexibleCameraMarkers } from './FlexibleCameraMarkers';
+import { moveCameraTargetTo, moveCameraTo } from './moveCamera';
+import { IFlexibleCameraManager } from './IFlexibleCameraManager';
+
+type RaycastCallback = (x: number, y: number, pickBoundingBox: boolean) => Promise<CameraManagerCallbackData>;
+
+/**
+ * Flexible implementation of {@link CameraManager}. The user can switch between Orbit, FirstPersion or OrbitInCenter
+ * Supports automatic update of camera near and far planes and animated change of camera position and target.
+ * @beta
+ */
+
+export class FlexibleCameraManager implements IFlexibleCameraManager {
+  //================================================
+  // INSTANCE FIELDS:
+  //================================================
+
+  private readonly _events = { cameraChange: new EventTrigger<CameraChangeDelegate>() };
+  private readonly _stopEventTrigger: DebouncedCameraStopEventTrigger;
+  private readonly _controls: FlexibleControls;
+  private readonly _camera: PerspectiveCamera;
+  private readonly _domElement: HTMLElement;
+  private readonly _inputHandler: InputHandler;
+  private readonly _markers?: undefined | FlexibleCameraMarkers;
+  private readonly _currentBoundingBox: Box3 = new Box3();
+  private _isDisposed = false;
+  private _nearAndFarNeedsUpdate = false;
+  private _isEnabled = true;
+
+  // For the wheel event
+  private _prevTime = 0;
+  private readonly _prevCoords = new Vector2();
+
+  //================================================
+  // INSTANCE FIELDS: Events
+  //================================================
+
+  private readonly _raycastCallback: RaycastCallback;
+
+  //================================================
+  // CONSTRUCTOR
+  //================================================
+
+  constructor(
+    domElement: HTMLElement,
+    inputHandler: InputHandler,
+    raycastCallback: RaycastCallback,
+    camera?: PerspectiveCamera,
+    scene?: Scene
+  ) {
+    this._camera = camera ?? new PerspectiveCamera(45, undefined, 0.1, 10000);
+    this._controls = new FlexibleControls(this.camera, domElement, new FlexibleControlsOptions());
+    this._domElement = domElement;
+    this._inputHandler = inputHandler;
+    this._raycastCallback = raycastCallback;
+    if (scene) {
+      this._markers = new FlexibleCameraMarkers(scene);
+    }
+
+    this.addEventListeners();
+    this._controls.addEventListeners(); // After this.addEventListeners();
+
+    this.controls.addEventListener('cameraChange', event => {
+      const { position, target } = event.camera;
+      this._events.cameraChange.fire(position.clone(), target.clone());
+      this._nearAndFarNeedsUpdate = true;
+      if (this._markers) {
+        this._markers.update(this);
+      }
+    });
```

Comment:
Is there a reason this can't be part of `this._controls.addEventListeners` (with some extra parameters)?